File: src/backend/joanie/core/serializers/course_run.py
"""CourseRun serializers"""

from rest_framework import serializers

from joanie.core import models

from .course import CourseSerializer


class CourseRunSerializer(serializers.ModelSerializer):
    """
    Serialize all information about a course run
    """

    course = CourseSerializer(read_only=True)

    class Meta:
        model = models.CourseRun
        fields = [
            "course",
            "end",
            "enrollment_end",
            "enrollment_start",
            "id",
            "resource_link",
            "start",
            "title",
            "state",
        ]
        read_only_fields = [
            "course",
            "end",
            "enrollment_end",
            "enrollment_start",
            "id",
            "resource_link",
            "start",
            "title",
            "state",
        ]

File: src/frontend/admin/src/services/api/models/CourseRun.ts
import type { Course, CourseState } from "./Course";

export type CourseRun = {
  id: string;
  title: string;
  course: Course;
  resource_link: string;
  start?: string | null;
  end?: string | null;
  enrollment_end?: string | null;
  enrollment_start?: string | null;
  languages: string[];
  is_gradable: boolean;
  is_listed: boolean;
  state?: CourseState;
  uri?: string;
};

export interface DTOCourseRun extends Omit<CourseRun, "course" | "id"> {
  id?: string;
  course_id: string;
}
